titiler/endpoints/cog.py

```diff
@@ -485,6 +486,15 @@ def wmts(
     )
 
 
+@router.get("/validate", response_class=JSONResponse)
+def validate_cog(
+    url: str = Query(..., description="Cloud Optimized GeoTIFF URL."),
+    strict: bool = Query(False, description="Treat warnings as errors"),
+):
+    """Validate a COG"""
+    return {"valid": cog_validate(url, strict=strict)}
```

Comment:
I think it would be helpful to return more output, such as whether windows exist. `rio-cogeo validate` allows for valid COG files to not have overviews, and just prints them as warnings.

Reply:
yeah, I'd love a mix of `rio cogeo info` and `rio cogeo validate`. sadly you can't run both directly. maybe we should refactor rio_cogeo.cogeo.validate to return the `info` needed for `rio cogeo info`

Reply:
Or `rio-cogeo info` to also call validate

Reply:
I'm not sure it's possible sadly. 
There is a `bug` in rasterio, once you call `rasterio.open('path', overview_level=1)` you can't open back the raw data. maybe this was fix 🤷‍♂️.

Reply:
well I just tested and it works 🤷‍♂️

Reply:
Once https://github.com/cogeotiff/rio-cogeo/pull/143/commits/7738517290fd3c0748b45b67594dfc2efa7c3168 and https://github.com/cogeotiff/rio-cogeo/pull/144 are released we should be good